project-base/storefront/helpers/serverSide/initServerSideProps.ts

```diff
@@ -157,7 +158,9 @@ export const initServerSideProps = async <VariablesType extends Variables>({
             // JSON.parse(JSON.stringify()) fix of https://github.com/vercel/next.js/issues/11993
             urqlState: JSON.parse(JSON.stringify(currentSsrCache.extractData())),
             isMaintenance,
-            cookiesStore: getCookiesStore(context),
+            cookiesStore: serializeCookieStoreOnServer(
+                createCookiesStoreOnServer(getCookiesStoreStateFromContext(context)).getState(),
+            ),
```

Comment:
Hmm, I wonder what's the point of this `serializeCookieStoreOnServer ` function? So far I see that logic of this utility can be simplified to 

```suggestion
            cookiesStore: createCookiesStoreOnServer(getCookiesStoreStateFromContext(context)).getState()
```

Reply:
Unfortunately, the state also contains the mutation function, which cannot be serialized, so if I just return that from the server as is, it fails.

Reply:
Actually, since I have removed the setter function, we do not need to serialize it manually and I was able to remove this function. So... resolved.